subsys/dfu/src/dfu_target_modem.c

```diff
@@ -0,0 +1,203 @@
+#include <zephyr.h>
+#include <flash.h>
+#include <nrf_socket.h>
+#include <net/socket.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(dfu_target_modem, CONFIG_DFU_TARGET_LOG_LEVEL);
+
+#define DIRTY_IMAGE 2621440
+#define MODEM_MAGIC 0x7544656d
+
+struct modem_delta_header {
+	u16_t pad1;
+	u16_t pad2;
+	u32_t magic;
+};
+
+static int  fd;
+static int  offset;
+
+static int get_modem_error(void)
+{
+	int rc;
+	int err = 0;
+	socklen_t len;
+
+	len = sizeof(err);
+	rc = getsockopt(fd, SOL_DFU, SO_DFU_ERROR, &err, &len);
+	if (rc) {
+		LOG_ERR("Unable to fetch modem error, errno %d", errno);
+	}
+
+	return err;
+}
+
+static int apply_modem_update(void)
+{
+	int err;
+
+	LOG_INF("Scheduling modem firmware update at next boot");
+
+	err = setsockopt(fd, SOL_DFU, SO_DFU_APPLY, NULL, 0);
+	if (err < 0) {
+		if (err == ENOEXEC) {
```

Comment:
socket APIs only return `0` or `-1`; you have to check `errno` when `err` is `-1`.
```
if (errno == ENOEXEC)
```

Reply:
fixed think this was a merge error :/